### PR TITLE
Update the utility function to add geoip info to a Heka message

### PR DESCRIPTION
**Pull request by trink**
Proposed use case:
https://github.com/mozilla-services/lua_sandbox_extensions/issues/220

### CODE REVIEW

File: geoip/io_modules/geoip/heka.lua [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+## Functions
+
+### add_geoip
+
+Given a Heka message add the geoip entries based on the specified field name.
+The requested geoip entries are specified in the `lookup` configuration table.
+The table consists of the entry name and the suffix to add to the field name,
+an empty suffix will overwrite the original value. See: the `opts` variable in
+https://github.com/agladysh/lua-geoip/blob/master/src/city.c for the available
+options, rommon values include "city" and "country_code".
```
Comment (ameihm0912) [inline geoip/io_modules/geoip/heka.lua]: Typo here `rommon -> common`

File: geoip/io_modules/geoip/heka.lua [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    if ptime + 3600 < ctime then
+        if geodb then geodb:close() end
+        geodb = assert(geoip.open(cfg.city_db_file))
+        ptime = ctime
+    end
+end
+
+
+local function validate_ip(ip)
+    local sl = #ip
```
Comment (ameihm0912) [inline geoip/io_modules/geoip/heka.lua]: Should we make sure `ip` is a string here?

File: geoip/io_modules/geoip/heka.lua [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+function add_geoip(msg, field_name)
+    if not msg.Fields then return end
+    local value = msg.Fields[field_name]
+    if not value or not validate_ip(value) then return end
+
+    refresh_db()
+
+    for k,v in pairs(cfg.lookup) do
+        local ret = geodb:query_by_addr(value, k)
```
Comment (ameihm0912) [inline geoip/io_modules/geoip/heka.lua]: Should this be wrapped in `pcall`? `query_by_addr` will fail on invalid value for `k`, potentially on invalid addresses

File: moz_security/tests/hindsight/run/input/generate_sshd.lua [hunk 4]
```diff
@@ -30,7 +30,7 @@ function process_message()
         msg.Uuid = v[1]
         msg.Fields.remote_user = v[2]
         msg.Fields.remote_addr = v[3]
-        msg = ga.geo_append(msg)
+        geo.add_geoip(msg, "remote_addr")
```
Comment (ameihm0912) [inline moz_security/tests/hindsight/run/input/generate_sshd.lua]: I'm guessing even though it's like this in the test, in the implementation we'd have the fields in the sandbox configuration (e.g., `fields = { "remote_addr", "etc" }`)

File: geoip/io_modules/geoip/heka.lua (hunk 3 above)
Comment (trink) [reply geoip/io_modules/geoip/heka.lua]: No if it is a bad k (configuration error) we should propagate the error)

File: geoip/io_modules/geoip/heka.lua (hunk 2 above)
Comment (trink) [reply geoip/io_modules/geoip/heka.lua]: Hmm, I am still debating on having this pre-validation at all (let me look at the geoip code and see if we can remove it all together)

File: moz_security/tests/hindsight/run/input/generate_sshd.lua (hunk 4 above)
Comment (trink) [reply moz_security/tests/hindsight/run/input/generate_sshd.lua]: No, this cannot be cfg driven (at least not from the geoip_heka cfg) as the field can be variable.  Some logs my have source_ip others remote_addr all handled by the same decoder. See the example in https://github.com/mozilla-services/lua_sandbox_extensions/issues/220

**Pull request merged**